src/uct/ib/base/ib_device.c

```diff
@@ -896,12 +899,16 @@ ucs_status_t uct_ib_device_select_gid(uct_ib_device_t *dev, uint8_t port_num,
                 goto out;
             }
 
-            if ((roce_prio[prio_idx].ver         == gid_info_tmp.roce_info.ver) &&
+            if (((ndev_name == NULL) ||
+                    (strncmp(ndev_name, gid_info_tmp.ndev_name, IFNAMSIZ) == 0)) &&
+                (roce_prio[prio_idx].ver         == gid_info_tmp.roce_info.ver) &&
```

Comment:
```suggestion
            if ((roce_prio[prio_idx].ver         == gid_info_tmp.roce_info.ver) &&
```

Please add at the beginning of the loop body:
```C
if (ucs_string_is_empty(ndev_name)) {
    uct_ib_device_get_roce_ndev_name(dev, port_num, i, ndev_name_tmp, sizeof(ndev_name_tmp))); 
    if (strcmp(ndev_name, ndev_name_tmp) != 0) {
        continue;
    }
}
```
,  you will also need local variable `char ndev_name_tmp[IFNAMSIZ]`.